### PR TITLE
Fix existing attributed links being removed when setting attributed text

### LINKED ISSUE #15: Setting an attributedText value containing link attributes does not work
Setting an attributedText value to the label that contains a `.link: URL` attribute does not work. This is because the `checkText()` method only looks at the raw string value of the attributedString for links. This means that links set that way do not work, they are not highlighted and are not clickable.

**Expected behavior**
Not sure if I should expect `NantesLabel` to handle this case but it would be good if it did.

### PR DESCRIPTION
**Pull request by spadafiva**
I think that this should resolve #15 at least in the case where I found an issue locally.

@chansen22 Let me know if this is workable for you.

### CODE REVIEW

**Comment by spadafiva** [comment]
@chansen22 Ready for another review.  I've updated to use Sets and added tests for the attributed string extensions and that the functionality works as expected

File: Nantes/NantesTests/NantesTests.swift [hunk 1]
```diff
@@ -215,6 +215,22 @@ final class NantesLabelTests: XCTestCase {
...
+        XCTAssertTrue(label.linkModels.isEmpty)
+        label.attributedText = linkString
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            promise.fulfill()
+        }
+        
+        XCTWaiter().wait(for: [promise], timeout: 2)
+        XCTAssertFalse(label.linkModels.isEmpty)
+    }
```
Comment (chansen22) [inline Nantes/NantesTests/NantesTests.swift]: Would you mind adding another test that adds multiple types of strings?

Something like:
```
    func testMultipleLinksAreAdded() {
        let testUrl = URL(string: "https://www.instacart.com")!
        let linkString = NSMutableAttributedString(string: "Link 1", attributes: [.link: testUrl])
        linkString.append(NSAttributedString(string: "https://www.swiftjoe.com"))
        XCTAssertTrue(label.linkModels.isEmpty)
        label.attributedText = linkString

        let promise = expectation(description: "waiting for attributes to be set")
        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
            promise.fulfill()
        }

        XCTWaiter().wait(for: [promise], timeout: 2)
        XCTAssertTrue(label.linkModels.count == 2)
    }
```

So we can test across both methods of pulling strings out too

File: Nantes/NantesTests/AttributedStringExtensionTests.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        XCTAssertFalse(existingLinks.isEmpty)
+    }
+    
+    func testFindsLinksForAttributedStringsCreatedFromHTML() {
+        let htmlString = "<a href=\"http://www.google.com\">I can be clicked\""
+        guard let data = htmlString.data(using: .utf8) else {
+            XCTFail()
+            return
+        }
+        do{
```
[the remark this answers is not in the document]
Comment (spadafiva) [reply Nantes/NantesTests/AttributedStringExtensionTests.swift]: 👍

File: Nantes/NantesTests/NantesTests.swift (hunk 1 above)
Comment (spadafiva) [reply Nantes/NantesTests/NantesTests.swift]: done

**Pull request merged**